### ISSUE TITLE
Note.derive not setting owner

### ISSUE THREAD

**Issue by naddison36** [AztecProtocol/aztec-v1#536]
## Expected Behaviour

When creating a note from an event log, the `owner` property should be set so it can be used in a JoinSplit proof.

```js
const note = require("aztec.js").note
const secp256k1 = require("@aztec/secp256k1")
const metadata = "0x1c6a72f049e538bb2c272529d626245da8e4380b3b78a848e7f35d54b6c4e7da27a387ec17338eefe6ab0c287807df0fd4437a3ad83190e0195e13505b609110dd6feecda50f75d733cf956c57e786976af334a9670c9dc27ce90139db3d1cf901"
const account = secp256k1.accountFromPrivateKey(
  "0x0000000000000000000000000000000000000000000000000000000000000003"
)
note.fromEventLog(
  metadata,
  account.privateKey
).then(note1 => {
  console.log(`Note owner ${note1.owner}`)
})
```

## Current Behaviour

The following error is thrown when the JoinSplit proof is sent to the zkAsset in a `confidentialTransfer` transaction
```
revert input note owner does not match
```

## Possible Solution

My current workaround is to explicitly set the `owner` property after the note is created and before the note is used in a JoinSplit proof. For example in the above, `note1.owner = account.address`. This allows the `confidentialTransfer` transaction to succeed.

I think the better solution is for `note.derive(spendingKey)` to set to `owner` property.
https://github.com/AztecProtocol/AZTEC/blob/bad7b4bd24a7bb07968e2c7b15af750b920618ec/packages/aztec.js/src/note/index.js#L116

## Your Environment

-   AZTEC packages and their versions: aztec.js 0.18.0
-   Node and yarn version: node v12.11.1

**Comment by naddison36** [comment]
@ArnSch can a new aztec.js package be published to npm with this fix? Thanks

**Comment by ArnSch** [comment]
:tada: This issue has been resolved in version 0.18.1 :tada:

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:

**Comment by ArnSch** [comment]
:tada: This issue has been resolved in version 1.7.1 :tada:

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:

**Comment by ArnSch** [comment]
:tada: This issue has been resolved in version 1.12.0 :tada:

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
